tests/components/tplink/test_number.py

```diff
@@ -0,0 +1,137 @@
+"""Tests for tplink number platform."""
+
+from kasa import Feature
+
+from homeassistant.components import tplink
+from homeassistant.components.number import (
+    ATTR_VALUE,
+    DOMAIN as NUMBER_DOMAIN,
+    SERVICE_SET_VALUE,
+)
+from homeassistant.components.tplink.const import DOMAIN
+from homeassistant.const import ATTR_ENTITY_ID, CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.setup import async_setup_component
+
+from . import (
+    DEVICE_ID,
+    MAC_ADDRESS,
+    _mocked_device,
+    _mocked_feature,
+    _mocked_strip_children,
+    _patch_connect,
+    _patch_discovery,
+)
+
+from tests.common import MockConfigEntry
+
+
+async def test_number(hass: HomeAssistant, entity_registry: er.EntityRegistry) -> None:
```

Comment:
We should be using snapshots here, to clearly see the full entity state & properties for each description defined on the platform.

Reply:
Yeah, the plan is to generalize the test code among all Feature-based platforms against the main PR to get it done properly for all platforms in a single sweep 👍